src/utils.rs

```diff
@@ -44,3 +44,23 @@ pub fn decode_base64(input: &str) -> Result<Vec<u8>, String> {
         base64::decode(input).map_err(|err| format!("invalid base64: {:?}", err))
     }
 }
+
+pub fn decode_decimal(input: &str) -> Result<Vec<u8>, String> {
+    input
+        .split_whitespace()
+        .map(|s| s.parse::<u8>().map_err(|err| format!("invalid decimal input: {:?}", err)))
+        .collect::<Result<Vec<u8>, String>>()
+}
+
+pub fn decode_binary(input: &str) -> Result<Vec<u8>, String> {
+    input
+        .split_whitespace()
+        .map(|s| {
+            if s.len() == 8 && s.chars().all(|c| c == '0' || c == '1') {
+                u8::from_str_radix(s, 2).map_err(|err| format!("invalid binary input: {:?}", err))
+            } else {
+                Err(format!("invalid binary input: not an 8-bit binary string"))
+            }
```

Comment:
here is the same request from me as in the prev comment. can you remove all chars except `0` and `1` and then try to parse it?

Reply:
Ack